include/soci/soci-backend.h

```diff
@@ -253,6 +256,64 @@ class blob_backend
     SOCI_NOT_COPYABLE(blob_backend)
 };
 
+class vector_blob_backend : public blob_backend
+{
+public:
+    std::size_t get_len() SOCI_OVERRIDE { return buffer_.size(); }
+
+    std::size_t read(std::size_t offset, char* buf,
+        std::size_t toRead) SOCI_OVERRIDE
+    {
+        toRead = (std::min)(toRead, buffer_.size() - offset);
+
+        // make sure that we don't try to read
+        // past the end of the data
+        memcpy(buf, buffer_.data() + offset, toRead);
+
+        return toRead;
+    }
+
+    std::size_t read_from_start(char* buf, std::size_t toRead,
+        std::size_t offset = 0) SOCI_OVERRIDE
+    {
+        return read(offset, buf, toRead);
+    }
+
+    std::size_t write(std::size_t offset, char const* buf,
+        std::size_t toWrite) SOCI_OVERRIDE
+    {
+        buffer_.resize((std::max)(buffer_.size(), offset + toWrite));
+
+        memcpy(buffer_.data() + offset, buf, toWrite);
+
+        return toWrite;
+    }
+
+    std::size_t write_from_start(const char* buf, std::size_t toWrite,
+        std::size_t offset = 0) SOCI_OVERRIDE
+    {
+        return write(offset, buf, toWrite);
+    }
+
+    std::size_t append(char const* buf, std::size_t toWrite) SOCI_OVERRIDE
+    {
+        return write(buffer_.size(), buf, toWrite);
+    }
+
+    void trim(std::size_t newLen) SOCI_OVERRIDE { buffer_.resize(newLen); }
+
+    std::size_t set_data(char const* buf, std::size_t toWrite)
+    {
+        buffer_.clear();
+        return write_from_start(buf, toWrite);
+    }
+
+    const char *get_buffer() const { return buffer_.data(); }
+
+protected:
+    std::vector< char > buffer_;
```

Comment:
`uint8_t` is not supported pre-C++11, as far as I know. So as it stands using this would currently fail the compilation as we don't yet require a modern C++ standard. I'll happily use `unsigned char` though. I only chose not to, as everywhere else that I looked, byte buffers in Soci seem to be treated as `char[]` and not `unsigned char[]`.

Reply:
Also the main `blob_backend` interface uses `char *` and not `unsigned char *` for the buffer.